Kudu.Core/Jobs/BaseJobRunner.cs

```diff
@@ -140,6 +185,10 @@ private void CacheJobBinaries(JobBase job, IJobLogger logger)
                     UpdateAppConfigs(tempJobInstancePath);
 
                     _workingDirectory = tempJobInstancePath;
+
+                    // cache the file map snapshot for next time (to aid in detecting
+                    // file deletions)
+                    _cachedSourceDirectoryFileMap = sourceDirectoryFileMap;
```

Comment:
shouldn't you write this to the file system, otherwise it seems meaningless (after restart)